### ISSUE TITLE
Sometimes Video Doesn't Connect

### ISSUE THREAD

**Issue by stephenlb** [stephenlb/webrtc-sdk#32]
Sometimes the stream does not establish.  This is a regression, v1 sdk does not have same issue.

**Comment by stephenlb** [comment]
I'm not sure if this is fixed or just started working...  :-/

**Comment by folkrav** [comment]
Last week I had some video that didn't connect for unknown reasons even with valid STUN/TURN servers. This may be the reason why? Not sure if it's related or not...

**Comment by stephenlb** [comment]
Yes there is a strange unknown state with V2 that causes this sometimes.

**Comment by stephenlb** [comment]
We are considering a v3 that will be more robust and have built-in resilience to this issue.
